conda_libmamba_solver/solver.py

```diff
@@ -479,12 +480,16 @@ def _specs_to_tasks_add(self, in_state: SolverInputState, out_state: SolverOutpu
                 # these are the EXPLICIT pins; conda also uses implicit pinning to
                 # constrain updates too but those can be overridden in case of conflicts.
                 if pinned.is_name_only_spec:
-                    # pins need to constrain in some way, otherwide is undefined behaviour
-                    pass
-                elif requested and not requested.match(pinned):
-                    # We don't pin; requested and pinned are different and incompatible,
-                    # requested wins and we let that happen in the next block
+                    # name-only pins are treated as locks when installed, see below
                     pass
+                elif requested:
+                    if out_state.attempt_count == 1:
+                        log.warning(
+                            "pinned spec %s is also user-requested. "
+                            "Overriding pinned spec with %s",
+                            pinned,
+                            requested,
+                        )
```

Comment:
Honestly, I would prefer if we didn't have this clause (i.e. we don't override the pin), which would mean we would be addressing https://github.com/conda/conda/issues/9016.